Cleanup the spec file

scarabeusiv opened the pull request:
As per commits first I detect libxml properly so we can build without hassle on Fedora.
I also by default told it to link with -fPIC and disable static libraries.

Then I cleanup the spec with spec-cleaner. It still builds on Fedora_24 and all the distributions enabled in the develprj like before.
Most notable change is that now the libstorage@LIBVERSION@ is able to be coinstalled with multiple versions. Translations were split to lang subpackage.

aschnell commented:
Closing since there was no reply from requester.

aschnell commented:
My review request was still pending and AFAIS thus not visible to others, sorry for that.

scarabeusiv commented:
Hopefully addressed the comment above.

FWIW The boost regex library could be also detected in configure.ac but as it is now at least erroring out if user ommits the headers by accident it should be quite okayish.

aschnell commented:
Fails in factory with 'ld: cannot find -lboost_unit_test_framework'. I suppose you need to add libboost_test-devel.

aschnell commented on libstorage-ng.spec.in:
```diff
@@ -15,246 +16,182 @@
+%package -n %{libname}
+Summary:        Library for storage management
+Group:          System/Libraries
+Requires:       %{name}-lang
```
Why not require the lang subpackage in the exact version?

scarabeusiv replied:
This is usually not done as it was considered fine to just pull the language, of course if you want just add the = %{version}

aschnell replied:
OK, I will likely add it someday.

aschnell commented:
LGTM
